packages/plugin-aws-elasticsearch/src/plugin.ts

```diff
@@ -0,0 +1,129 @@
+import assert = require('assert')
+import { resolve } from 'path'
+import retry = require('bluebird-retry')
+import Bluebird = require('bluebird')
+import Errors = require('common-errors')
+import Elasticsearch = require('@elastic/elasticsearch')
+import { NotFoundError } from 'common-errors'
+import { PluginTypes, Microfleet } from '@microfleet/core'
+import * as AWS from 'aws-sdk'
+import createAwsElasticsearchConnector from './utils/createAwsElasticsearchConnector'
+import { PluginInterface, ValidatorPlugin } from '@microfleet/core/lib'
+
+interface ElasticLogger {
+  trace(...args: any[]): void;
+  debug(...args: any[]): void;
+  info(...args: any[]): void;
+  warning(...args: any[]): void;
+  error(...args: any[]): void;
+  fatal(...args: any[]): void;
+  close(...args: any[]): void;
+}
+
+export interface AWSElasticsearchConfig {
+  node: string;
+  accessKeyId: string;
+  secretAccessKey: string;
+  region?: string;
+  log: any;
+}
+
```

Comment:
add module augmentation so that types work when using the plugin, please look at other plugins for examples

```ts
declare module '@microfleet/core-types' {
   interface Microfleet {
     awsElasticsearch: ...
  }

  interface ConfigurationOptional {
    ...
  }
}
```